### ISSUE TITLE
Log flood with "Two methods with same score(0)"

### ISSUE THREAD

**Issue by rsandoz** [orphan-oss/ognl#42]
Our logs are being flooded with the message:
Two methods with same score(0): "public java.lang.String myPackage.MyClass.myMethod(myPackage.MyModelClass,int)" and "public abstract java.lang.String myPackage.MyClassAbstract.myMethod(java.lang.Object,int)" please report!

We are on the latest version of struts (struts2-core-2.5.14.jar) which uses ognl-3.1.15.

This issue seems to come from OgnlRuntime.java at line 1430.

I grepped on future versions of OgnlRuntime.java from your github for 'abstract' ignoring case and found code in 3.1.16 and 3.2.3, and 3.2.4.   I tried applying the logic from these code fixes in a locally compiled OgnlRuntime, but the problem did not go away.  I don't completely understand your score calculation, but some of the other flags (isInterface, etc) seem to preclude our implementation.

The only thing I could do to workaround this was to put above this sysout line:
if (!Modifier.isAbstract(m.getModifiers()))

Our implementation is:
<<MyClassAbstract.java>>
package myPackage;
public abstract class MyClassAbstract <E> {
    public abstract String myMethod (E element, int i);

<<MyClass.java>>
package myPackage;
public class MyClass extends MyClassAbstract < MyModelClass > {
    public String myMethod (MyModelClass myModel, int i) {

Please advise.

**Comment by lukaszlenart** [comment]
You can try to use [OGNL 3.1.16](https://github.com/jkuhnert/ognl/blob/master/docs/VersionNotes.md#release-notes---version-3116-324)

**Comment by rsandoz** [comment]
That was one of the first things we tried.  It didn't work.  Still had the log flood with said message.  

I actually compiled OgnlRuntime from 3.1.15 and put the class in my classpath ahead of the one baked into the jar.  In doing so, I could comment out the message, which I prefer not to do.

I tried incorporating just the abstract logic from future revisions in my custom copy of 3.1.15 OgnlRuntime.java and could not get it.

I tried looking at the source for 3.1.16 OgnlRuntime.java.  It seems that looking at lines 1804, 1806, and 1838, the method, isDefaultMethod, which is where it checks to see if the modifier, ABSTRACT, only get used if isInterface is set.  I tried  removing this check, but that didn't fix it.  I even tried incorporating the logic going from 3.2.3 to 3.2.4 where 3.2.4 has the  same logic as 3.1.16 (the isDefaultMethod method) 3.2.3 collects the interfaces and continues the loop if isAbstract is set (line 2138)
